geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/commands/DeleteFeatureCommand.scala

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2015 Commonwealth Computer Research, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.locationtech.geomesa.tools.commands
+
+import com.beust.jcommander.{JCommander, Parameters, ParametersDelegate}
+import com.typesafe.scalalogging.slf4j.Logging
+import org.locationtech.geomesa.tools.commands.DeleteFeatureCommand.DeleteFeatureParams
+
+import scala.util.{Failure, Success, Try}
+
+class DeleteFeatureCommand(parent: JCommander) extends CommandWithCatalog(parent) with Logging {
+  override val command = "deletefeature"
+  override val params = new DeleteFeatureParams
+
+  override def execute() = {
+    if (Option(params.pattern).isEmpty && Option(params.featureName).isEmpty) {
+      throw new IllegalArgumentException("Error: Must provide either featureName or pattern to delete")
+    }
+
+    val features = getFeatureList()
+    validate(features) match {
+      case Success(_) =>
+        if (params.force || promptConfirm(features)) {
+          removeAll(features)
+        } else {
+          logger.info(s"Cancelled deletion")
+        }
+      case Failure(ex) =>
+        println(s"Feature validation failed on error: ${ex.getMessage}")
```

Comment:
(picky) The output messages in this file are sentences, and yet, do not end with punctuation  Unsatisfying